contracts/ColonyStorage.sol

```diff
@@ -37,6 +37,22 @@ contract ColonyStorage is DSAuth, DSMath {
   address colonyNetworkAddress;
   ERC20Extended token;
 
+  struct TokenIssuanceRate {
+    // Amount describing how many tokens can be issued for one month
+    uint256 amount;
+    // Last time the token issuance rate is changed
+    // It is used to assure that rate can be changed once every 4 weeks
+    uint256 timestamp;
+  }
+
+  // Keeps information about token issuance rate amount and last change timestamp
+  TokenIssuanceRate tokenIssuanceRate;
+  // Amount of tokens allowed in circulation
+  uint256 tokenSupplyCeiling;
+  // Keeps track of how many tokens are issued in terms of seconds
+  // TODO: If we decide to mint entire available amount in `mintTokens`, there will be no need for this variable
+  uint256 totalAmountIssuedUntilTimestamp;
```

Comment:
I wonder if we can make this implementation really clean by also moving this `totalAmountIssuedUntilTimestamp` to the `TokenIssuanceRate` struct params and calling it `totalAmountIssuedUnderRate`

Reply:
This way there will be 2 added variables to Colony: 1 for the supply ceiling and the other for the issuance rate. Nice and clean. I suspect the logic that you've already refactored quite well will become even simpler.